pkg/oras/settings.go

```diff
@@ -24,18 +26,33 @@ type orasSettings struct {
 	tag                string
 }
 
-// parseContainer URI into a container and a tag
-func (settings *orasSettings) parseReference(reference string) {
-	container := strings.TrimPrefix(reference, "oras://")
+// parseContainer URI into a reference(registry+repository) and a tag
+func (settings *orasSettings) parseReference(reference string) error {
+	artifactRef := strings.TrimPrefix(reference, "oras://")
 	tag := "latest"
-	if strings.Contains(container, ":") {
-		parts := strings.Split(container, ":")
-		container = parts[0]
-		tag = parts[1]
+
+	// if the refeence doesn't have a / then add a docker.io/library prefix
+	// this is to support the docker.io library that is implicit
+	// in docker pull commands
+	if !strings.Contains(artifactRef, "/") {
+		artifactRef = "docker.io/library/" + artifactRef
+	}
+
+	ref, err := registry.ParseReference(artifactRef)
+	if err != nil {
+		return err
 	}
+
+	// Oras uses the ref to mean digest or tag
```

Comment:
That's not confusing or anything! :laughing: It continues to make me chuckle how overloaded these terms are!

Reply:
Long story (but this is because of how the OCI layout doesn't have a contect of a tag)